### PR TITLE
[DOPS-1029] switch to jenkins library

### PR DESCRIPTION
**Pull request by BulatSaif**
Switch to jenkins library, we do not need complicated logic with bakong anymore

For more information see : https://github.com/soramitsu/nbc-deploy/pull/22

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,99 +1,5 @@
...
-                            env.TAG = dockerTags[env.BRANCH_NAME]
-                            sh "./gradlew dockerPush"
-                        }
-                        withCredentials([usernamePassword(credentialsId: 'nexus-nbc-deploy', usernameVariable: 'DOCKER_REGISTRY_USERNAME', passwordVariable: 'DOCKER_REGISTRY_PASSWORD')]) {
-                            env.DOCKER_REGISTRY_URL = "https://nexus.iroha.tech:19000"
-                            env.TAG = dockerTags[env.BRANCH_NAME]
-                            sh "./gradlew dockerPush"
-                        }
-                    } else if (env.TAG_NAME ==~ /^bakong-\d{1,4}\.\d.*/) {
-                        def tagPattern = (env.TAG_NAME =~ /(?<=bakong-)(\d{1,4}\.\d.*)/)[0][1]
```
Comment (ravenolf) [inline Jenkinsfile:64]: Don't we need to handle this case anymore ?

**Review by ravenolf — APPROVED** [review]

File: Jenkinsfile [hunk 2]
```diff
@@ -1,99 +1,5 @@
...
-    post {
-        always {
-            publishHTML (target: [
-                allowMissing: false,
-                alwaysLinkToLastBuild: false,
-                keepAll: true,
-                reportDir: 'build/reports',
-                reportFiles: 'd3-test-report.html',
-                reportName: "D3 test report"
-            ])
```
Comment (Cre-eD) [inline Jenkinsfile:93]: Do we need this report?

File: Jenkinsfile (hunk 1 above)
Comment (BulatSaif) [reply Jenkinsfile:64]: no, handling of this case moved to https://github.com/soramitsu/nbc-deploy/pull/22

File: Jenkinsfile (hunk 2 above)
Comment (BulatSaif) [reply Jenkinsfile:93]: I will return it back

**Review by kogeler — APPROVED** [review]
OK

File: Jenkinsfile (hunk 2 above)
Comment (BulatSaif) [reply Jenkinsfile:93]: @Mingela , told what they did not use it. 
We will configure universal junit xml reports for all repos later.

**Review by Mingela — APPROVED** [review]

**Pull request merged**